daml-lf/spec/daml-lf-1.rst

```diff
@@ -1413,121 +1413,117 @@ need to be evaluated further. ::
      ⊢ᵥ  λ x : τ . e
 
-   ——————————————————————————————————————————————————— ValExpTyAbs
+   ——————————————————————————————————————————————————— ValTyAbs
      ⊢ᵥ  Λ α : k . e
 
-   ——————————————————————————————————————————————————— ValExpLitInt64
+   ——————————————————————————————————————————————————— ValLitInt64
      ⊢ᵥ  LitInt64
 
-   ——————————————————————————————————————————————————— ValExpLitNumeric
+   ——————————————————————————————————————————————————— ValLitNumeric
      ⊢ᵥ  LitNumeric
 
-   ——————————————————————————————————————————————————— ValExpLitText
+   ——————————————————————————————————————————————————— ValLitText
      ⊢ᵥ  LitText
 
-   ——————————————————————————————————————————————————— ValExpLitDate
+   ——————————————————————————————————————————————————— ValLitDate
      ⊢ᵥ  LitDate
 
-   ——————————————————————————————————————————————————— ValExpLitTimestamp
+   ——————————————————————————————————————————————————— ValLitTimestamp
      ⊢ᵥ  LitTimestamp
 
-   ——————————————————————————————————————————————————— ValExpLitContractId
+   ——————————————————————————————————————————————————— ValLitContractId
      ⊢ᵥ  cid
 
-   ——————————————————————————————————————————————————— ValExpUnit
+   ——————————————————————————————————————————————————— ValUnit
      ⊢ᵥ  ()
 
-   ——————————————————————————————————————————————————— ValExpTrue
+   ——————————————————————————————————————————————————— ValTrue
      ⊢ᵥ  'True'
 
-   ——————————————————————————————————————————————————— ValExpFalse
+   ——————————————————————————————————————————————————— ValFalse
      ⊢ᵥ  'False'
 
-   ——————————————————————————————————————————————————— ValExpListNil
+   ——————————————————————————————————————————————————— ValListNil
      ⊢ᵥ  'Nil' @τ
 
-     ⊢ᵥ  e₁     ⊢ᵥ  e₂
-   ——————————————————————————————————————————————————— ValExpListCons
+     ⊢ᵥ  eₕ     ⊢ᵥ  eₜ
+   ——————————————————————————————————————————————————— ValListCons
      ⊢ᵥ  'Cons' @τ eₕ eₜ
 
-   ——————————————————————————————————————————————————— ValExpListNil
+   ——————————————————————————————————————————————————— ValListNil
      ⊢ᵥ  'None' @τ
 
      ⊢ᵥ  e
-   ——————————————————————————————————————————————————— ValExpListCons
+   ——————————————————————————————————————————————————— ValListCons
      ⊢ᵥ  'Some' @τ e
 
-     ⊢ᵥ  e₁      …      ⊢ᵥ  eₙ
-   ——————————————————————————————————————————————————— ValExpRecCon
-     ⊢ᵥ  Mod:T @τ₁ … @τₙ { f₁ = e₁, …, fₙ = eₙ }
-
-     1 ≤ k ≤ m
+     0 ≤ k < m
      𝕋(F) = ∀ (α₁: ⋆) … (αₘ: ⋆). σ₁ → … → σₙ → σ
-   ——————————————————————————————————————————————————— ValExpBuiltin₁
+   ——————————————————————————————————————————————————— ValBuiltin₁
      ⊢ᵥ  F @τ₁ … @τₖ
 
-     1 ≤ k < n
+     0 ≤ k < n
      𝕋(F) = ∀ (α₁: ⋆) … (αₘ: ⋆). σ₁ → … → σₙ → σ
      ⊢ᵥ  e₁      …      ⊢ᵥ  eₖ
-   ——————————————————————————————————————————————————— ValExpBuiltin₂
+   ——————————————————————————————————————————————————— ValBuiltin₂
      ⊢ᵥ  F @τ₁ … @τₘ e₁ … eₖ
 
```

Comment:
Assume the 4 builtin  functions with the following type: 
```
   𝕋(F₀₀) =  σ
   𝕋(F₁₀) = ∀ α₁. σ
   𝕋(F₀₁) = σ₁ → σ
   𝕋(F₁₁) = ∀ α₁. σ₁ → σ
```
  
And the followin proposition 
 


00.  `⊢ᵥ F₀₀` must **not** hold 
  This was not holding before.  
  This does not hold after.

10.  `⊢ᵥ F₁₀` must hold, 
    *This was not holding before.* 
    This hold after using  `ValBuiltin₂` with `k=0`
  
01.  `⊢ᵥ  F₀₁` must hold,
    *This was not holding before.* 
    This hold after using  `ValBuiltin₂` with `k=0`

11.   `⊢ᵥ  F₁₁` must hold,  
      This was holding before using  `ValExpBuiltin₁` with k = 1
      This hold after using ValBuiltin₂ with k = 0

110.  `⊢ᵥ   F₁₀ @τ` must **not** hold
     This was holding before using  `ValExpBuiltin₁` with k = 1
     This does not hold after

111.  `⊢ᵥ F₁₁ @τ` must hold,
     This was holding before using  `ValExpBuiltin₁` with k = 1
     This hold after using `ValBuiltin₂` with k = 0